src/complex/DataStructure/EmptyDataStore.hpp

```diff
@@ -202,6 +206,34 @@ class EmptyDataStore : public AbstractDataStore<T>
     throw std::runtime_error("");
   }
 
+  /**
+   * @brief Creates and returns an EmptyDataStore from the provided DatasetReader
+   * @param datasetReader
+   * @return std::unique_ptr<EmptyDataStore>
+   */
+  static std::unique_ptr<EmptyDataStore> readHdf5(const H5::DatasetReader& datasetReader)
+  {
+    // tupleShape
+    H5::AttributeReader tupleShapeAttribute = datasetReader.getAttribute(AbstractDataStore<T>::k_TupleShape);
```

Comment:
Since they tuple and component shape reading is the same between `DataStore` and `EmptyDataStore`, it could be moved into a separate function used by both.
e.g. `std::vector<IDataStore::ShapeType> ReadTupleShape(const H5::DatasetReader& datasetReader)`